src/golang/lib/execution_state/utils.go

```diff
@@ -0,0 +1,134 @@
+package execution_state
```

Comment:
Let's avoid naming file 'utils' esp. when there's a clearer scope available. For example, we can call this file 'integration'. This [post](https://breadcrumbscollector.tech/stop-naming-your-python-modules-utils/) explains some motivation on using more specific naming to improve code organization.